### PR TITLE
Update Sphinx theme

### PR DESCRIPTION
**Pull request by arnaucasau**
### Summary
Updates to new Ecosystem theme.

For more information: https://github.com/Qiskit/qiskit_sphinx_theme/issues/232

### CODE REVIEW

**Comment by kevinsung** [comment]
Thank you for the pull request! The CI failures look unrelated. I will investigate.

**Comment by arnaucasau** [comment]
Thank you @kevinsung ! I'm changing the pull request to a draft momentarily to test a little the build first.

**Comment by kevinsung** [comment]
CI should be fixed now, please merge main

File: docs/index.rst [hunk 1]
```diff
@@ -2,6 +2,11 @@
 Qiskit Research documentation
 =============================
 
+.. toctree::
```
Comment (kevinsung) [inline docs/index.rst:5]: What does this do? Does it matter that there is already another `toctree` directive below?

Comment (arnaucasau) [reply docs/index.rst:5]: This toctree adds an entry only visible in the left side menu called Home. The toctree below continues working and doing the same as before (displaying the content in the left side menu and in the page).

Comment (kevinsung) [reply docs/index.rst:5]: Ok.

**Comment by kevinsung** [comment]
@arnaucasau is this ready to merge?

**Comment by arnaucasau** [comment]
Yes, it's ready to merge :)

**Review by kevinsung — APPROVED** [review]

**Pull request merged**